frontend/src/lib/components/FilterBar/FilterBar.svelte

```diff
@@ -67,6 +67,7 @@
     watcher = () => debouncer.current;
   }
   watch(watcher, (value) => {
+    if ($allFilters[searchKey] === value) return;
```

Comment:
Does this actually make a difference? I would hope it doesn't 😄

Reply:
I believe the `allFilters` store would fire if any index/property got updated, because stores are not as fine-grained as Svelte 5 state. So this avoids a mistaken firing when it wasn't needed.

Reply:
This is using `watch` (on a simple string), not `$effect`, so I don't think that's the case, because `allFilters` is not a dependency here.

Reply:
I didn't mean that allFilters would cause the watch to re-run. I meant that when `$allFilters[searchKey]` gets assigned to in the next line, all subscribers of the `allFilters` score would get notified that it had a new value, not just the subscribers that care about that particular search key, because `$allFilters` contains an object and Svelte stores don't try to compare objects by value, they just fire off a subscriber notification if the object MIGHT have been changed. (If it was a state, then it would have gotten fine-grained reactivity and each individual prop would be keeping track of its own subscribers, so the values being compared for subscriber notifications would be strings and then this "if the value was already this one" comparison wouldn't be necessary).